File: work/abc042_d.cpp
#include <iostream>
#include <vector>
using namespace std;

using ull = unsigned long long;

vector<ull> g,g2;
ull x = 1;

ull pow(ull a, ull n){
  if(n==0){
    return 1;
  }else if(n==1){
    return a;
  }else if(n%2==0){
    ull t = pow(a, n/2);
    return (t*t)%x;
  }else{
    return (a*pow(a, n-1))%x;
  }
}

ull route(int _h, int _w){
  int h = _h-1;
  int w = _w-1;
  ull ret = 1;
  return (((g[h+w]*g2[h])%x)*g2[w])%x;
}

int main() {
  int H,W,A,B;
  cin>>H>>W>>A>>B;
  for(int i=0;i<9;++i){
    x *= 10;
  }
  x += 7;

  g.push_back(1);
  g.push_back(1);
  g2.push_back(1);
  g2.push_back(1);

  for(int i=2;i<=H+W;++i){
    g.push_back((g[i-1]*i)%x);
    g2.push_back(pow(g[i], x-2));
    // cout<<g[i]<<" "<<g2[i]<<endl;
  }

  ull ret = 0;
  for(int h=1;h<H-A+1;++h){
    ull tmp = route(h, B);
    tmp *= route(H-h+1, W-B);
    ret = (ret+tmp)%x;
  }
  cout<<ret;
}

File: work/abc042_d.py
# coding=utf-8
from functools import lru_cache

line = input().split(" ")
H = int(line[0])
W = int(line[1])
A = int(line[2])
B = int(line[3])

g = [1, 1]
for i in range(2,H+W):
    g.append(g[i-1]*i)

def fac(n):
    if n in g:
        return g[i]
    print("calc!")
    ret = 1
    for i in range(1,n+1):
        ret *= i
    return ret

def route(_h,_w):
    h = _h-1
    w = _w-1
    ret = 1

    return fac(h+w)/(fac(w)*fac(h))

ret = 0
x = 10**9+7
for h in range(1,H-A+1):
    tmp = route(h, B)
    tmp *= route(H-h+1, W-B)

    ret = (ret+tmp)%x

print(int(ret))
